torch/csrc/jit/codegen/cuda/ir_nodes.cpp

```diff
@@ -2886,6 +2887,16 @@ bool NamedScalar::sameAs(const Statement* other) const {
   return other->as<NamedScalar>()->name().compare(name()) == 0;
 }
 
+bool NamedScalar::isTensorSize() const {
+  static const std::regex r(R"(T\d+\.size\[\d+\])");
+  return std::regex_match(name(), r);
```

Comment:
This would be fine, but why don't we add static methods to create tensor size and stride named scalars and explicitly tag them as such? That seems more robust than the string-based pattern match.

Reply:
I generally agree, but I prefer to keep `NamedScalar` string only. We could have new subclasses of `Val` for values that has special meaning, for example `class TensorAttribute : public Val`. And I believe a good place to do this refactor is at https://github.com/csarofeen/pytorch/pull/2282, where I need to access the `.data` of a global tensor.

Reply:
Yeah, that would be better, and it should have a dependency connection to the tensor itself. I just don't like the string matching-based approach as it seems backward.